### PR TITLE
Add com.github.matfantinel.reminduck

### PR DESCRIPTION
**Pull request by matfantinel**

### CODE REVIEW

File: com.github.matfantinel.reminduck.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    "command": "com.github.matfantinel.reminduck",
+    "finish-args": [
+       "--share=ipc",
+       "--share=network",
+       "--socket=x11",
+       "--socket=wayland",
+       "--talk-name=org.gtk.vfs.*",
+       "--talk-name=ca.desrt.dconf",
+       "--filesystem=home",
+       "--filesystem=xdg-run/dconf",
```
Comment (bilelmoussaoui) [inline com.github.matfantinel.reminduck.json]: You should remove dconf access. It's not needed anymore starting from FDO Sdk 19.08/GNOME Sdk 3.34

File: com.github.matfantinel.reminduck.json [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+       "*.la",
+       "*.a"
+    ],
+    "modules": [
+       {
+          "name": "libyaml",
+          "sources": [
+              {
+                  "type": "git",
+                  "url": "https://github.com/yaml/libyaml.git",
```
Comment (bilelmoussaoui) [inline com.github.matfantinel.reminduck.json]: Use a tarball or add a commit along with the tag. The same for the other modules.

**Comment by bilelmoussaoui** [comment]
bot, build com.github.matfantinel.reminduck

File: com.github.matfantinel.reminduck.json [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+              }
+          ]
+       },
+       {
+          "name": "reminduck",
+          "buildsystem": "meson",
+          "sources": [
+              {
+                  "type": "git",
+                  "url": "https://github.com/matfantinel/reminduck.git",
```
Comment (bilelmoussaoui) [inline com.github.matfantinel.reminduck.json]: You probably want to patch the application to use the elementary GTK & icons themes. They are already in the BaseApp.

Comment (matfantinel) [reply com.github.matfantinel.reminduck.json]: I've tested it with many different themes in a Gnome environment and am happy with the result. I prefer it to respect GTK theming!

File: com.github.matfantinel.reminduck.json (hunk 2 above)
Comment (bilelmoussaoui) [reply com.github.matfantinel.reminduck.json]: Please don't mark conversations as resolved unless they are.

**Comment by bilelmoussaoui** [comment]
The build fails due to missing OARS tag in the appdata file https://hughsie.github.io/oars/generate.html

**Comment by matfantinel** [comment]
Sorry @bilelmoussaoui , I've updated the tags and the manifest now.

**Comment by LeandroStanger** [comment]
bot, build com.github.matfantinel.reminduck

File: com.github.matfantinel.reminduck.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    "runtime": "org.gnome.Platform",
+    "runtime-version": "3.34",
+    "sdk": "org.gnome.Sdk",
+    "base": "io.elementary.BaseApp",
+    "base-version": "juno-19.08",
+    "command": "com.github.matfantinel.reminduck",
+    "finish-args": [
+       "--share=ipc",
+       "--share=network",
+       "--socket=x11",
```
Comment (bilelmoussaoui) [inline com.github.matfantinel.reminduck.json]:
```suggestion
       "--socket=fallback-x11",
```

**Comment by bilelmoussaoui** [comment]
bot, build com.github.matfantinel.reminduck

File: com.github.matfantinel.reminduck.json [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+              "-Dstemming=false"
+          ],
+          "sources": [
+              {
+                  "type": "git",
+                  "url": "https://github.com/ximion/appstream.git",
+                  "tag": "v0.12.9",
+                  "commit": "d77cc0f331ff5d3bd43be104106a94ed201c62ec"
+              }
+          ]
```
Comment (bilelmoussaoui) [inline com.github.matfantinel.reminduck.json]: Include lmdb as a submodule of appstream https://github.com/flathub/io.github.NhekoReborn.Nheko/blob/master/io.github.NhekoReborn.Nheko.json#L28

**Comment by bilelmoussaoui** [comment]
bot, build com.github.matfantinel.reminduck

**Comment by bilelmoussaoui** [comment]
I have tried the app and noticed that two icons are missing from the default icon theme
Screenshot from 2020-01-10 23-45-12

Screenshot from 2020-01-10 23-44-55

File: com.github.matfantinel.reminduck.json [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    "base": "io.elementary.BaseApp",
+    "base-version": "juno-19.08",
+    "command": "com.github.matfantinel.reminduck",
+    "finish-args": [
+       "--share=ipc",
+       "--share=network",
+       "--socket=fallback-x11",
+       "--socket=wayland",
+       "--talk-name=org.gtk.vfs.*",
+       "--filesystem=home"
```
Comment (bilelmoussaoui) [inline com.github.matfantinel.reminduck.json]: Couldn't find why you would need home access, network or vfs?

File: com.github.matfantinel.reminduck.json [hunk 7]
```diff
@@ -0,0 +1,87 @@
+{
+    "app-id": "com.github.matfantinel.reminduck",
+    "runtime": "org.gnome.Platform",
```
Comment (bilelmoussaoui) [inline com.github.matfantinel.reminduck.json]: Any reason for not using the freedesktop runtime here?

Comment (matfantinel) [reply com.github.matfantinel.reminduck.json]: I have based my manifest on another app, and don't really know the difference between the two

**Comment by matfantinel** [comment]
About the missing icons... I suppose the best way to deal with that is by forcing the elementary icons within the app like you mentioned. I'll have to research on how to do that and will update this PR when possible.

Thanks a lot for the feedback!

**Comment by Exalm** [comment]
Not trying to persuade, but there are also some issues with the style. Like look at header labels, which use the default style because Adwaita doesn't have granite style classes. Or the headerbar height, and you can see the back button is awkwardly aligned. (that's caused by `default-decoration` style class FTR). Basically things like this is why elementary apps are often suggested to hardcode stylesheet, as trying to support both is often a huge amount of work.

File: com.github.matfantinel.reminduck.json (hunk 7 above)
Comment (bilelmoussaoui) [reply com.github.matfantinel.reminduck.json]: All the runtimes/sdk's are based on freedesktop runtime/sdk. The GNOME runtime/sdk contains libraries and tools specific to GNOME like Vala. As you're already using the elementary baseApp which contains vala + granite & the elementary stylesheet & icons, you can just switch to the freedesktop runtime.

**Comment by matfantinel** [comment]
Thanks everyone for your input. I have updated the application and manifest to use the freeDesktop SDK and runtime, removed unnecessary permissions, and patched in the elementary theme. I tested it locally and all boxes seemed to check.

**Comment by LeandroStanger** [comment]
bot, build com.github.matfantinel.reminduck

**Pull request closed without merging**